test/unit/helpers/feature-layer-metadata.spec.js

```diff
@@ -256,6 +256,7 @@ describe('FeatureLayerMetadata', () => {
       metadata: {
         foo: 'bar',
         displayField: 'myField',
+        copyrightText: 'Custom copyright text',
```

Comment:
Since `copyrightText` override is handled in lib/helpers/table-layer-metadata.js, this test should be added to table-layer-metadata.spec.js rather than feature-layer-metadata.spec.js.

Reply:
Before I make this change, I want to say that having thought about it some more, it seems wiser to delete this test entirely and put the test into the higher up `layer-metadata.spec.js` instead (which I'll comment on separately there).

In any event, since unit tests are a kind of safety net against future refactorings, it seems wiser to put the test where its intention is clear (putting copyrightText on a feature layer). The fact that table layers get the functionality for free is irrelevant (no one cares)-- it is an implementation detail. (Should I move the implementation down into feature-layer to be clear that this is the intention?)

However, if in the future, you (or the maintainer at that time if it is not you) decide that the abstraction of a feature layer being a type of table layer is the wrong abstraction, moving this test will lose all the value of the protection against that refactoring potentially introducing a bug where the feature layer (due to refactoring) no longer has the text that needs to appear in the attribution on the webmap.

That said, I'm happy to comply with your request.

Reply:
> Before I make this change, I want to say that having thought about it some more, it seems wiser to delete this test entirely and put the test into the higher up `layer-metadata.spec.js` instead (which I'll comment on separately there).

Thanks for your thoughts, but I disagree with this.  In my opinion that would be not be a unit test, because `layer-metadata.js` is really only responsible for instantiating the classes, and not doing anything with `copyrightText` at all.  It would make it more of an integration test.  

> However, if in the future, you (or the maintainer at that time if it is not you) decide that the abstraction of a feature layer being a type of table layer is the wrong abstraction moving this test will lose all the value of the protection against that refactoring potentially introducing a bug

Certainly possible, and a good point.  Still, I call what you're asking for as integration test coverage, which you are welcome to add. We do have some integration tests, but layer-info requests aren't currently covered.

Reply:
I moved this into integration tests. At first, I thought I would try to move all the tests, but quickly realized it was too ambitious, so I did the least reasonably possible while still at least adding this one test case.